code/datums/components/food/edible.dm

```diff
@@ -250,28 +250,33 @@ Behavior that's still missing from this component that original food items had t
 		else
 			this_food.reagents.add_reagent(r_id, amount)
 
+///Makes sure the thing hasn't been destroyed or fully eaten to prevent eating phantom edibles
+/datum/component/edible/proc/IsFoodGone(var/atom/owner, mob/living/feeder)
+	if(QDELETED(owner)|| !(IS_EDIBLE(owner)))
+		return TRUE
+	if(!owner.reagents.total_volume)//Shouldn't be needed but it checks to see if it has anything left in it.
+		to_chat(feeder, "<span class='warning'>None of [owner] left, oh no!</span>")
+		if(isturf(owner))
+			var/turf/T = owner
+			T.ScrapeAway(1, CHANGETURF_INHERIT_AIR)
+		else
+			qdel(owner)
```

Comment:
Same with this, why is this handled in `IsFoodGone` which doesn't imply that it mutates the world?